api/cas/cas_auth.py

```diff
@@ -249,8 +250,30 @@ def wrap(*args, **kwargs):
             if authorized is not None:
                 return wrapped_function(*args, **kwargs)
 
-        abort(403, description="Not authorized.")
+        abort(status.HTTP_403_FORBIDDEN, description="Not authorized.")
 
     return wrap
 
 
+def edl_federated_request(url, stream_response=False):
+    s = requests.Session()
+    response = s.get(url, stream=stream_response)
+
+    if response.status_code == status.HTTP_401_UNAUTHORIZED:
```

Comment:
If this is `False`, this function returns `None` instead of the `response` from the line preceding the `if`.

Reply:
I removed the extraneous if conditions from this method